src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,286 @@
+
+    This function calls '_build_a_list', '_build_b_list', and '_flag_ent_exit'
+    in order to fully form a_list and b_list. The 'a_list' and 'b_list'
+    that it returns are the fully updated vectors of PolyNodes that represent
+    'poly_a' and 'poly_b', respectively. This function also returns
+    'a_idx_list', which at its "ith" index stores the index in 'a_list' at
+    which the "ith" intersection point lies. This function also returns
+    'intr_list', which at the "ith" index contains the cartesian coordinates
+    for the "ith" intersection point. Finally, this function returns 'edges_a'
+    and 'edges_b' which are vectors of the cartesian points of the original
+    polygons, 'poly_a' and 'poly_b'.
+"""
+function _build_ab_list(poly_a, poly_b)
+    # Make a list for nodes of each polygon. Note the definition of PolyNode
+    a_list = Array{PolyNode, 1}(undef, _nedge(poly_a))
+    b_list = Array{PolyNode, 1}(undef, _nedge(poly_b))
+
+    # Initialize arrays to keep track of the important information 
+    # associated with the intersection points of poly_a and poly_b.
+    # I initialize these arrays assumed there will be a maximum of 
+    # 30 intersection points and then I truncate the arrays later.
+    k = 4
+    # intr_list stores the cartesian coordintes of the intersection point
+    intr_list = Array{Tuple{Real, Real}, 1}(undef, k)
+    # At index i, a_idx_list stores the index of the ith intersection point in a_list
+    a_idx_list = Array{Int, 1}(undef, k)
+    # At index i, b_idx_list stores the edge number of poly_b that the ith intersection
+    # point lies on.
+    b_idx_list = Array{Int, 1}(undef, k)
+    # Alpha values are used to determine the order in which to place
+    # intersection points in a polygon list, which is especially
+    # useful when many intersection points lie on the same edge of a polygon.
+    alpha_a_list = Array{Real, 1}(undef, k)
+    alpha_b_list = Array{Real, 1}(undef, k)
+
+    # These lists store the cartesian coordinates of poly_a and poly_b
+    edges_a = to_edges(poly_a)
+    edges_b = to_edges(poly_b)
+
+    intr_list, a_idx_list, b_idx_list, 
+    alpha_a_list, alpha_b_list = _build_a_list(edges_a, edges_b, intr_list, a_idx_list, b_idx_list, 
+                                                        alpha_a_list, alpha_b_list, a_list, b_list)
+    b_neighbors, b_list = _build_b_list(b_list, intr_list, a_idx_list, b_idx_list, alpha_b_list)
+
+    # Iterate through a_list and update the neighbor indices
+    for ii in eachindex(a_list)
+        if a_list[ii].inter
+            a_list[ii].neighbor = b_neighbors[a_list[ii].idx]
+        end
+    end
+
+    # Flag the entry and exists
+    a_list, b_list = _flag_ent_exit(GI.trait(poly_a), poly_a, GI.trait(poly_b), poly_b, a_list, b_list, edges_a, edges_b)
+
+    return a_list, b_list, a_idx_list, intr_list, edges_a, edges_b
+end
+
+# This is the struct that makes up a_list and b_list
+mutable struct PolyNode
```

Comment:
I think we should add in a point tuple `(x, y)`. I know we were avoiding this, but I think it needlessly complicates things and requires us to have `edges_a` and `edges_b`, which I think we can get rid of. 

I think we can also make `alpha` into `fracs` and have it be a tuple `(alpha, beta)`, which will let us get rid of `alpha_b_list`. 

Finally, both of those tuples should have type `::Tuple{T, T}`, and then the PolyNode can be `mutable struct PolyNode{T} where T <: AbstractFloat`

Reply:
Can also get rid of `idx`.